pkg/sourcesystem/gitSourceSystem.go

```diff
@@ -1,16 +1,54 @@
 package sourcesystem
 
 import (
-	"fmt"
+	"errors"
 	"os/exec"
+
+	git "gopkg.in/libgit2/git2go.v23"
 )
 
-func gitAddSource() {
-	cmd := exec.Command("git", "clone", "--depth", "2", "testRepo", "testPath")
+// GitVersionControl is the git implemenation
+// of the SourceControlManager
+type GitVersionControl struct{}
+
+// CloneSource implements a git shallow clone of depth 2
+func (scm GitVersionControl) CloneSource(repo *SourceRepository, location string) error {
+	cmd := exec.Command("git", "clone", "--depth", "2", repo.SourceOrigin, location)
+
+	cmdErr := cmd.Start()
+	if cmdErr != nil {
+		// TODO: need dev logs here
```

Comment:
Need to have two different sets of logs. First set sends all system level issues that we need as developers for debugging to the file log. The second set (not yet implemented) needs to send the success/error messages back to the end user (such as when a clone is successful). Still need to think about this and the second system is not in place yet in the logger.